scripts/install.sh

```diff
@@ -569,4 +569,7 @@ function main() {
     report_install_success
 }
 
-main "$@"
+# tee logs into /var/log/kurl/install-<date>.log and stdout
+mkdir -p /var/log/kurl
+main "$@" 2>&1 | tee -a /var/log/kurl/install-$(date +"%m-%d-%Y-%H-%M-%S").log
+
```

Comment:
I liked how simple this looks! Three things caught my attention though:

1. Maybe we should use a directory in `/var/lib/kurl` to store the log (not strong opinion here);
2. We loose access to `stderr`, now all is `stdout`. Not sure if this impacts anything;
3. We loose the return code of the `main` function (now it is always zero), aren't all testgrid become green because of that ?

As I don't know if these are real problems I prefer to hear what other have to say.

Reply:
Hi @ricardomaraschini,

Thank you for the help and review. Following the answers.

> 1. Maybe we should use a directory in /var/lib/kurl to store the log (not strong opinion here);

/var/lib -> has only libs for the system
/var/log -> has the logs for the system. This dir has logs for other application and etc. The approach to create the dir with the name of the app from where the logs came from can also be checked there. 

> We loose access to stderr, now all is stdout. Not sure if this impacts anything;

Not sure, I could check all been outputting as usual and the same content in the logs
 
> We loose the return code of the main function (now it is always zero), aren't all testgrid become green because of that ?

I do not think so, because in the testgrid we have post scripts to check the results right. 
But that is a great point for we have attention. We can merge and check it out.

Reply:
Hi @ricardomaraschini,

> 3. We loose the return code of the main function (now it is always zero), aren't all testgrid become green because of that ?

See the second commit : https://github.com/replicatedhq/kURL/pull/3891/commits/78f3281b9cf3e7ed22c8734cf64a72cc4fe14f39

Now the script will finish with the error. I did a test with:

<img width="589" alt="Screenshot 2022-12-22 at 15 34 49" src="https://user-images.githubusercontent.com/7708031/209168835-92547407-02d5-4cda-b95f-6b0af47b7a7d.png">

Reply:
> We loose the return code of the main function (now it is always zero), aren't all testgrid become green because of that ?

We don't thanks to `set -o pipefail` (line number 5 above)

Reply:
@ricardomaraschini I need to remove that option but see that we can only use tee.